README.md

```diff
@@ -386,73 +386,40 @@ Try to match the expression. If the match does not succeed, just return
 
 #### & { _predicate_ }
 
-The predicate is a piece of JavaScript code that is executed as if it was inside
-a function. It gets the match results of labeled expressions in preceding
-expression as its arguments. It should return some JavaScript value using the
-`return` statement. If the returned value evaluates to `true` in boolean
-context, just return `undefined` and do not consume any input; otherwise
-consider the match failed.
-
-The code inside the predicate can access all variables and functions defined in
-the initializer at the beginning of the grammar.
+This is a positive assertion. No input is consumed.
 
-The code inside the predicate can also access location information using the
-`location` function. It returns an object like this:
+The predicate should be JavaScript code, and it's executed as a
+function. Curly braces in the predicate must be balanced.
 
-```javascript
-{
-  source: options.grammarSource,
-  start: { offset: 23, line: 5, column: 6 },
-  end: { offset: 23, line: 5, column: 6 }
-}
-```
+The predicate should `return` a boolean value. If the result is
+truthy, the match result is `undefined`, otherwise the match is
+considered failed.
 
-The `start` and `end` properties both refer to the current parse position. The
-`offset` property contains an offset as a zero-based index and `line` and
-`column` properties contain a line and a column as one-based indices.
-
-The code inside the predicate can also access options passed to the parser using
-the `options` variable.
-
-Note that curly braces in the predicate code must be balanced.
+The predicate has access to all variables and functions in the
+[Action Execution Environment](#action-execution-environment).
 
 #### ! { _predicate_ }
 
-The predicate is a piece of JavaScript code that is executed as if it was inside
-a function. It gets the match results of labeled expressions in preceding
-expression as its arguments. It should return some JavaScript value using the
-`return` statement. If the returned value evaluates to `false` in boolean
-context, just return `undefined` and do not consume any input; otherwise
-consider the match failed.
+This is a negative assertion. No input is consumed.
 
-The code inside the predicate can access all variables and functions defined in
-the initializer at the beginning of the grammar.
+The predicate should be JavaScript code, and it's executed as a
+function. Curly braces in the predicate must be balanced.
 
-The code inside the predicate can also access location information using the
-`location` function. It returns an object like this:
+The predicate should `return` a boolean value. If the result is
+falsy, the match result is `undefined`, otherwise the match is
```

Comment:
s/the/it's/ like above, if you decide that's right.

Reply:
I'm not a native speaker, so I trust you :)

Reply:
Done